kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/NettyFilterDispatchExecutor.java

```diff
@@ -49,7 +49,7 @@ EventLoop getEventLoop() {
     @Override
     public <T> CompletionStage<T> completeOnFilterDispatchThread(CompletionStage<T> completionStage) {
         Objects.requireNonNull(completionStage, "completionStage was null");
-        CompletableFuture<T> future = new InternalCompletableFuture<>(this);
+        CompletableFuture<T> future = new InternalCompletableFuture<>(eventLoop);
         completionStage.whenComplete((value, throwable) -> {
             if (isInFilterDispatchThread()) {
                 forward(value, throwable, future);
```

Comment:
presumably, the first formal parameter of (forward) should be marked `@Nullable`, as futures can be completed with `null`.